contracts/party/PartyGovernanceNFT.sol

```diff
@@ -356,33 +373,29 @@ contract PartyGovernanceNFT is PartyGovernance, ERC721, IERC2981 {
         {
             IERC20 prevToken;
             for (uint256 i; i < withdrawTokens.length; ++i) {
-                IERC20 token = withdrawTokens[i];
-
                 // Check if order of tokens to transfer is valid.
                 // Prevent null and duplicate transfers.
-                if (prevToken >= token) revert InvalidTokenOrderError();
+                if (prevToken >= withdrawTokens[i]) revert InvalidTokenOrderError();
 
-                prevToken = token;
+                prevToken = withdrawTokens[i];
 
                 // Check token's balance.
-                uint256 balance = address(token) == ETH_ADDRESS
+                uint256 balance = address(withdrawTokens[i]) == ETH_ADDRESS
                     ? address(this).balance
-                    : token.balanceOf(address(this));
+                    : withdrawTokens[i].balanceOf(address(this));
 
                 // Add fair share of tokens from the party to total.
                 for (uint256 j; j < tokenIds.length; ++j) {
                     // Must be retrieved before burning the token.
-                    uint256 shareOfVotingPower = getVotingPowerShareOf(tokenIds[j]);
-
-                    withdrawAmounts[i] += (balance * shareOfVotingPower) / 1e18;
+                    withdrawAmounts[i] += (balance * getVotingPowerShareOf(tokenIds[j])) / 1e18;
```

Comment:
wouldn't the previous version be preferred to reduce MLOADs?

Reply:
> wouldn't the previous version be preferred to reduce MLOADs?

Will cause stack overflow